### PR TITLE
[PropertyInfo] Mentioned the phpDoc collection type

### LINKED ISSUE #10338: [PropertyInfo] Implement "Collection" types in PhpDocExtractor
https://github.com/symfony/symfony/pull/26300

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #10338.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@popy-dev I know that you implemented this feature a long ago (thanks for that!) but, could you please review if this proposed change is correct? Thanks.

File: components/property_info.rst [hunk 1]
```diff
@@ -315,9 +315,15 @@ a collection - a non-scalar value capable of containing other values. Currently
...
+  is ``array``;
 * The mutator method the property is derived from has a prefix of ``add``
-  or ``remove`` (which are defined as the list of array mutator prefixes).
+  or ``remove`` (which are defined as the list of array mutator prefixes);
+* The phpDocumentor annotation is of type "collection" (e.g.
+  ``@var Iterator<DateTime>``, ``@var \Iterator<integer,string>``, etc.)
+
+.. versionadded:: 4.2
+
+    The support of phpDocumentor collection types was introduced in Symfony 4.2.
```
Comment (OskarStark) [inline components/property_info.rst]: Should we add a link to the phpDocumentor library?

**Comment by popy-dev** [comment]
Hi @javiereguiluz 

It seems fine to me. Maybe you could give more "open" example (the Iterator class can, in fact, be any class, and collection values also), but I guess that's nitpicking :)

**Comment by javiereguiluz** [comment]
@popy-dev thanks for reviewing!

I have some problems trying to fully understand this feature so ... could you please show me some valid examples that don't use the `Iterator` class? Thanks!

**Comment by popy-dev** [comment]
> I have some problems trying to fully understand this feature

The idea is to be able to give more details about a property type. If you have an array of Items, you can use the `@var Item[]` notation to give more information than just `@var array`. But this notation does not covers the case of nexted arrays, neither the case of Traversable objects. That's what the notation `@var Class<key,value>`covers. It's expected that "Class" implements the Traversable interface, so it could be any class implementing Iterator, or IteratorAggregate, or it could be a Generator. And as it's mostly a matter of convention, one could argue that it does not even need to be a Traversable. I hope I'm clearer than it looks :D

As an actual example, if you're familiar with doctrine collection, you could annotate a OneToMany collection with :

```
/**
 * @var Doctrine\Common\Collections\Collection<App\Entity\Item>
 */
protected $items;
```

**Comment by javiereguiluz** [comment]
@popy-dev thanks a lot for your great and detailed explanation. I can fully understand this now and I've updated the explanation to show better examples. Thanks!

**Pull request merged**